src_multispec/MassFluxUtil.cpp

```diff
@@ -1,49 +1,190 @@
-				BL_TO_FORTRAN_ANYD(rho[mfi]),
-				BL_TO_FORTRAN_ANYD(rhotot[mfi]),
-				BL_TO_FORTRAN_ANYD(molarconc[mfi]),
-				BL_TO_FORTRAN_ANYD(molmtot[mfi]));
+        const Array4<const Real>& rho = rho_in.array(mfi);
+        const Array4<const Real>& rhotot = rhotot_in.array(mfi);
+        const Array4<      Real>& molarconc = molarconc_in.array(mfi);
+        const Array4<      Real>& molmtot = molmtot_in.array(mfi);
+        
+
+        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        {
+
+            GpuArray<Real, MAX_SPECIES> RhoN;
+            GpuArray<Real, MAX_SPECIES> MolarConcN;
+
+            for (int n=0; n<nspecies; ++n ){
+                RhoN[n] = rho(i,j,k,n);
+            }
+
+
+            ComputeMolconcMolmtotLocal(nspecies, molmass, 
+                            RhoN, rhotot(i,j,k),          
+                            MolarConcN, molmtot(i,j,k));
+
+            for (int n=0; n<nspecies; ++n ){
+                molarconc(i,j,k,n) = MolarConcN[n] ;
+            }
+
+        });
+
     }
 
 }
 
-void ComputeGamma(const MultiFab& molarconc,
-		  const MultiFab& Hessian,
-		  MultiFab& Gamma)
+// EP
+void ComputeGamma(const MultiFab& molarconc_in,
+		      const MultiFab& Hessian_in,
+          MultiFab& Gamma_in)
+//void ComputeGamma(const MultiFab& molarconc,
+//		  const MultiFab& Hessian,
+//		  MultiFab& Gamma)
 {
   
     BL_PROFILE_VAR("ComputeGamma()",ComputeGamma);
 
-    int ng = Gamma.nGrow();
+    //int ng = Gamma.nGrow();
+    int ng = Gamma_in.nGrow();
     
     // Loop over boxes
-    for (MFIter mfi(Gamma,TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+    //for (MFIter mfi(Gamma,TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+    for (MFIter mfi(Gamma_in,TilingIfNotGPU()); mfi.isValid(); ++mfi) {
 
         // Create cell-centered box
         const Box& bx = mfi.growntilebox(ng);
 
-        compute_Gamma(ARLIM_3D(bx.loVect()), ARLIM_3D(bx.hiVect()),
-		      BL_TO_FORTRAN_ANYD(molarconc[mfi]),
-		      BL_TO_FORTRAN_ANYD(Hessian[mfi]),
-		      BL_TO_FORTRAN_ANYD(Gamma[mfi]));
+        //compute_Gamma(ARLIM_3D(bx.loVect()), ARLIM_3D(bx.hiVect()),
+		    //  BL_TO_FORTRAN_ANYD(molarconc[mfi]),
+		    //  BL_TO_FORTRAN_ANYD(Hessian[mfi]),
+		    //  BL_TO_FORTRAN_ANYD(Gamma[mfi]));
+
+        //EP-Starts here
+        const Array4<const Real>& molarconc = molarconc_in.array(mfi);
+        const Array4<const Real>& Hessian = Hessian_in.array(mfi); //What if I want a 6-dim array?
+        const Array4<      Real>& Gamma = Gamma_in.array(mfi);
+
+        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        {
+
+            GpuArray<Real, MAX_SPECIES> MolarConcN;
+            //GpuArray<Real, MAX_SPECIES> HessianN;
+            //GpuArray<Real, MAX_SPECIES> GammaN;
+            Array2D<Real, MAX_SPECIES, MAX_SPECIES*MAX_SPECIES> GammaN;
+            Array2D<Real, MAX_SPECIES, MAX_SPECIES*MAX_SPECIES> HessianN;
```

Comment:
I'm guessing you either want to use an Array2D<Real, MAX_SPECIES, MAX_SPECIES>, or GpuArray<Real, MAX_SPECIES*MAX_SPECIES>